### PR TITLE
[ issue54 ] Add Pagination To Books And Favorites Page

### LINKED ISSUE #54: Add Pagination To Books And Favorite Pages
**Is your feature request related to a problem? Please describe.**
N/A

**Describe how the solution should behave**
pagination to be added so that users are not overloaded with data and they can navigate by using pagination

**Screenshots of the design/if any**
Screenshot 2021-10-17 182135
**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Niyaz-Dev12**

### CODE REVIEW

File: src/components/pagination/Pagination.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    } else if (currentButton > numberOfPages.length - 3) {
+      // > 7
+      const sliced = numberOfPages.slice(numberOfPages.length - 4); // slice(10-4)
+      tempNumberOfPages = [1, dotsLeft, ...sliced];
+    } else if (currentButton === dotsInitial) {
+      // [1, 2, 3, 4, "...", 10].length = 6 - 3  = 3
+      // arrOfCurrButtons[3] = 4 + 1 = 5
+      // or
+      // [1, 2, 3, 4, 5, "...", 10].length = 7 - 3 = 4
+      // [1, 2, 3, 4, 5, "...", 10][4] = 5 + 1 = 6
```
Comment (MuhannedNoman) [inline src/components/pagination/Pagination.jsx]: Remove any unused code to reduce noise in code.

File: src/components/pagination/Pagination.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      tempNumberOfPages = [
+        1,
+        dotsLeft,
+        ...sliced1,
+        ...sliced2,
+        dotsRight,
+        numberOfPages.length,
+      ]; // [1, '...', 4, 5, 6, '...', 10]
+    } else if (currentButton > numberOfPages.length - 3) {
+      // > 7
```
Comment (MuhannedNoman) [inline src/components/pagination/Pagination.jsx]: Remove comment doesn't add anything.

File: src/components/pagination/Pagination.jsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    if (numberOfPages.length < 6) {
+      tempNumberOfPages = numberOfPages;
+    } else if (currentButton >= 1 && currentButton <= 3) {
+      tempNumberOfPages = [1, 2, 3, 4, dotsInitial, numberOfPages.length];
+    } else if (currentButton === 4) {
+      const sliced = numberOfPages.slice(0, 5);
+      tempNumberOfPages = [...sliced, dotsInitial, numberOfPages.length];
+    } else if (currentButton > 4 && currentButton < numberOfPages.length - 2) {
+      // from 5 to 8 -> (10 - 2)
```
Comment (MuhannedNoman) [inline src/components/pagination/Pagination.jsx]: Remove

**Review by MuhannedNoman — APPROVED** [review]
LGTM, just clean the comments and unused code and feel free to merge.

**Comment by Niyaz-Dev12** [comment]
@MuhannedNoman the comments are for clarification but i will remove it

**Pull request merged**